### ISSUE TITLE
We need to improve the print dialog

### ISSUE THREAD

**Issue by timlinux** [inasafe/inasafe#4705]
### Problem

The print dialog is confusing and should be more flexible. Hopefully my wireframe diagram below explains it all. I would like to include this for the 4.3 release because the current situation isnt good. Current:

![screen shot 2017-11-30 at 12 26 52](https://user-images.githubusercontent.com/178003/33426251-dbe0fa7e-d5c9-11e7-811b-547490548b56.png)

Proposed:

![screen shot 2017-11-30 at 12 00 30](https://user-images.githubusercontent.com/178003/33426172-9bc97952-d5c9-11e7-97b2-00e256766874.png)

**Comment by timlinux** [comment]
CC @Charlotte-Morgan

**Comment by Charlotte-Morgan** [comment]
Hi @timlinux - happy to make this a bit richer.
Could you also look at the open composer dialog - am I the only person who finds its not intuitive?

**Comment by timlinux** [comment]
Suggestion from @Charlotte-Morgan to maybe only enable landscape map by default

Selected options should be saved for next use of dialog

**Comment by Charlotte-Morgan** [comment]
Hi @myarjunar 
There is a typo fro you - the word is provenance... I also have a python error when I try this - will try some more
![image](https://user-images.githubusercontent.com/5743135/33868100-194f98d6-df55-11e7-841a-e76c616b7068.png)

**Comment by Charlotte-Morgan** [comment]
Python error when I choose my own template
```
An error has occurred while executing Python code: 

AttributeError: 'PrintReportDialog' object has no attribute 'selected_template_path' 
Traceback (most recent call last):
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\gui\tools\print_report_dialog.py", line 540, in accept
    components = self.prepare_components()
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\gui\tools\print_report_dialog.py", line 495, in prepare_components
    if not exists(self.selected_template_path):
AttributeError: 'PrintReportDialog' object has no attribute 'selected_template_path'


Python version: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)] 
QGIS version: 2.18.14 Las Palmas, ff83b9a479 

Python Path:
C:/PROGRA~1/QGIS2~1.18/apps/qgis/./python/plugins\processing
C:/PROGRA~1/QGIS2~1.18/apps/qgis/./python
C:/Users/AIFDR_ADMIN/.qgis2/python
C:/Users/AIFDR_ADMIN/.qgis2/python/plugins
C:/PROGRA~1/QGIS2~1.18/apps/qgis/./python/plugins
C:\Program Files (x86)\ArcGIS\Desktop10.2\bin
C:\Program Files (x86)\ArcGIS\Desktop10.2\arcpy
C:\Program Files (x86)\ArcGIS\Desktop10.2\ArcToolbox
C:\PROGRA~1\QGIS2~1.18\bin\python27.zip
C:\PROGRA~1\QGIS2~1.18\apps\Python27\DLLs
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\plat-win
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\lib-tk
C:\PROGRA~1\QGIS2~1.18\bin
C:\PROGRA~1\QGIS2~1.18\apps\Python27
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\jinja2-2.7.2-py2.7.egg
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\markupsafe-0.23-py2.7-win-amd64.egg
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\win32
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\win32\lib
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\Pythonwin
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\Shapely-1.2.18-py2.7-win-amd64.egg
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\wx-2.8-msw-unicode
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\xlrd-0.9.2-py2.7.egg
C:\PROGRA~1\QGIS2~1.18\apps\Python27\lib\site-packages\xlwt-0.7.5-py2.7.egg
C:/Users/AIFDR_ADMIN/.qgis2//python
C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe
C:\Users\AIFDR_ADMIN\.qgis2\python\plugins\inasafe\safe_extras\parameters
C:\Users\AIFDR_ADMIN\.qgis2\python\plugins
C:\Users\AIFDR_ADMIN\.qgis2\python\plugins\inasafe\safe_extras
C:/AIFDR II - SDA/10_InaSAFE Analysis/20171120 Jakarta 4.3
```

**Comment by Charlotte-Morgan** [comment]
Hi @myarjunar 
I am not sure this is working as you might expect. When I check only 2 options like this![image](https://user-images.githubusercontent.com/5743135/33868178-94e55fb2-df55-11e7-8af8-8c8fe0f870aa.png)
I end up with the provenance report, action checklist and the impact report.

I have asked @felix-yew to test some more. I am reopening this ticket

**Comment by felix-yew** [comment]
Hi @timlinux @Charlotte-Morgan @myarjunar - This modification I have try this issue and try some experience.
1. For `Tabular report` - If I have run the report before and try print in the same result layer but in different checked report, the generated report also appear.
2. It is good to have an explanation in Help button. But it will better if explain more about template override and/or where the location for it (option 3-`Custom and map report`). What/Where is the search directories means (in option 4-`Custom and map report`). If it include in folder `.qgis` or else, please explain it more to help the user know.
3. I also get an error when try open composer in `Custom and map report` in option 5 -
![image](https://user-images.githubusercontent.com/13621886/33874976-44bf1dc2-df53-11e7-98ec-e6ab46d4fb9f.png)
```
2017-12-12T14:26:12 1 Traceback (most recent call last):
File "C:/Users/yanuaf/.qgis2/python/plugins\inasafe\safe\gui\tools\print_report_dialog.py", line 540, in accept
components = self.prepare_components()
File "C:/Users/yanuaf/.qgis2/python/plugins\inasafe\safe\gui\tools\print_report_dialog.py", line 495, in prepare_components
if not exists(self.selected_template_path):
AttributeError: 'PrintReportDialog' object has no attribute 'selected_template_path'
```

**Comment by ismailsunni** [comment]
Hi @felix-yew 
I have checked and tested, your issues are solved in #4775, can you try it also? If you are happy, we can close this ticket.
For non-fatal error or only polishing suggestion, please add it in #4776

**Comment by felix-yew** [comment]
Hi mas @ismailsunni @myarjunar - I have try the PR :+1: :+1: LGTM
